t04_refactored.py

```diff
@@ -193,13 +193,63 @@ def team_11_adv(username, delay):
     pass
     # TODO Add your code here
     return True
-###################################################################################
-
-
+#################################################################################
 def team_12_adv(username, delay):
-    pass
-    # TODO Add your code here
+    """
+    https://docs.google.com/document/d/1-VIVi-Txn87fIW0pf9LwcYZbWpGTJC0tbpnC-va0IBo/edit?usp=sharing
+    Besher Kitaz
+    Feda Mohammadi
+    :return: None
+    """
+    direction = input("[Tavern/Inn/Reform/Run]")
+
+    if direction == "Inn":
+        # Good choice!
+        print("""You wander into an Inn, where a gambling man at a corner table offers you 
+              a gold nugget in exchange for playing a game of Russian Roulette.""")
+        sleep(delay)
+        print("Upon closer inspection you realize the revolver's chambers are all full. "
+              "Regardless, you choose to entertain the idea.")
+        print()
+    if direction == "Tavern":
+        # Good choice!
+        print("Although you referred to the building as a tavern initially, closer inspection reveals to you"
+              "that this particular subset of business is referred to as a 'saloon' in a western context.")
+        sleep(delay)
+        print()
+    if direction == "Saloon":
```

Comment:
This should be `elif`

Reply:
Also, `Saloon` wasn't an option given to the user to pick.

Reply:
Added this to the options in the input prompt. Changed if to elif.